File: internal-base/mvps/src/main/java/com/lijing/dev/mvps/test/MainActivity.java
package com.lijing.dev.mvps.test;

import android.view.View;
import android.widget.Button;

import com.lijing.dev.mvps.R;
import com.lijing.dev.mvps.activity.BaseAbstractActivity;
import com.lijing.dev.mvps.di.DaggerBaseMvpsComponent;

public class MainActivity extends BaseAbstractActivity<MainPresenter> {


    @Override
    public int getContentLayoutID() {
        return R.layout.activity_main;
    }

    @Override
    public void initVariables() {

    }

    @Override
    public void initViews() {
        Button button = findViewById(R.id.button);
        button.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                mPresenter.show("123123");
            }
        });
    }

    @Override
    public void initEvents() {
    }

    @Override
    public void initInject() {
        DaggerBaseMvpsComponent.getInstance().inject(this);
    }

    @Override
    public void initData() {
    }
}

File: internal-base/mvvm/src/main/java/com/lijing/dev/mvvm/test/MainActivity.kt
package com.lijing.dev.mvvm.test

import android.widget.Button
import com.lijing.dev.mvvm.R
import com.lijing.dev.mvvm.core.BaseAbstractActivity
import com.lijing.dev.mvvm.di.DaggerMvvmComponent
import com.lijing.dev.network.di.BaseApiComponent

class MainActivity : BaseAbstractActivity<MainViewModel>() {

    override fun getContentLayoutID(): Int {
        return R.layout.activity_main
    }

    override fun bindViewModel() {
        DaggerMvvmComponent.builder()
                .baseApiComponent(BaseApiComponent.getInstance())
                .build().inject(this)
    }

    override fun initVariables() {
    }

    override fun initViewsAndEvents() {
        findViewById<Button>(R.id.btn_show)
                .setOnClickListener { viewModel.doSomeThing() }
    }

}
